shuffle-coordinator/src/test/java/com/alibaba/flink/shuffle/coordinator/metrics/MetricsTest.java

```diff
@@ -0,0 +1,160 @@
+import com.alibaba.flink.shuffle.core.config.MemoryOptions;
+import com.alibaba.flink.shuffle.core.config.RestOptions;
+import com.alibaba.flink.shuffle.core.config.StorageOptions;
+import com.alibaba.flink.shuffle.core.config.TransferOptions;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for metrics. */
+public class MetricsTest {
+
+    private final Random random = new Random();
+
+    private final int maxRetries = 10;
+
+    private CloseableHttpClient httpClient;
+
+    @Before
+    public void before() throws Exception {
+        httpClient = HttpClients.createDefault();
+    }
+
+    @After
+    public void after() throws Exception {
+        if (httpClient != null) {
+            httpClient.close();
+        }
+    }
+
+    @Test
+    public void testGetManagerMetrics() throws Exception {
+        try (ShuffleManagerEntrypoint manager = createShuffleManager()) {
+            manager.start();
+            Configuration configuration = manager.getConfiguration();
+            int port = configuration.getInteger(RestOptions.REST_MANAGER_BIND_PORT);
+            HttpGet request = new HttpGet("http://127.0.0.1:" + port + "/metrics");
+
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                assertEquals(200, response.getStatusLine().getStatusCode());
+                String metricsJson = EntityUtils.toString(response.getEntity());
+                assertTrue(metricsJson.contains("\"jvm\":"));
+                assertTrue(metricsJson.contains("\"system\":"));
+            }
+        }
+    }
+
+    @Test
+    public void testGetWorkerMetrics() throws Exception {
+        try (ShuffleWorkerRunner worker = createShuffleWorker()) {
+            Configuration configuration = worker.getConfiguration();
+            int port = configuration.getInteger(RestOptions.REST_WORKER_BIND_PORT);
+            HttpGet request = new HttpGet("http://127.0.0.1:" + port + "/metrics");
+
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                assertEquals(200, response.getStatusLine().getStatusCode());
+                String metricsJson = EntityUtils.toString(response.getEntity());
+                assertTrue(metricsJson.contains("\"jvm\":"));
+                assertTrue(metricsJson.contains("\"system\":"));
+                assertTrue(metricsJson.contains("remote-shuffle"));
+            }
+        }
+    }
+
+    @Test
+    public void testGetGroupMetrics() throws Exception {
```

Comment:
It seems that the mode of group = A & group = B is allowed. Can you add a test for this.